libafl/src/state/mod.rs

```diff
@@ -137,6 +137,26 @@ pub trait HasMetadata {
     {
         self.metadata().get::<M>().is_some()
     }
+
+    /// To get metadata
+    fn get_metadata<M>(&self) -> Result<&M, Error>
+    where
+        M: SerdeAny,
+    {
+        self.metadata().get::<M>().ok_or_else(|| {
+            Error::key_not_found(format!("{} not found", core::any::type_name::<M>()))
+        })
+    }
+
+    /// To get mutable metadata
+    fn get_mut_metadata<M>(&mut self) -> Result<&mut M, Error>
```

Comment:
Should probably be named `get_metadata_mut` to follow naming conventions (it's always mut at the end, unless if the type is specifically named MutSomething)